In [ ]:
import sys 
import os
current_dir = os.getcwd()
if os.path.basename(current_dir) == "experiments":
    os.chdir("..")
    
project_root = os.getcwd()

if project_root not in sys.path:
    sys.path.append(project_root)

from src.utils.load_settings import load_yaml
from src.retrieval.retrieval_orchestrator import run_retrieval


config = load_yaml.__wrapped__("config/config.yaml")

12:38:02 | Loading yaml config | INFO | YAML config loaded successfuly from config/config.yaml


In [ ]:
def show_chunks(prompt, top=config["retriever"]["top_k"], threshold=config["retriever"]["distance_treshold"]):
    context, raw_chunks, score = run_retrieval(user_query=prompt, emb_model=config["models"]["embedding_model"], db_path="chroma_db_old", retriever_top_k=top, distance_threshold=threshold)

    print(f"Znaleziono chunków: {len(raw_chunks)}\n")
    for i, doc in enumerate(raw_chunks):
        print(f"CHUNK {i+1} | Score {score[i]}")
        print("-" * 60)
        print(doc)
        print("=" * 60)

In [24]:
show_chunks("What is the official title of the financial report for IFC's 2024 fiscal year?")
# 0.194

08:39:30 | Retrieval | INFO | Starting retrieval process.
08:39:32 | Retrieval | INFO | Found 20 chunks meeting the distance threshold (<= 0.6).
08:39:32 | Retrieval | INFO | Successfully post-processed 20 chunks. Best score: 0.194


Znaleziono chunków: 20

CHUNK 1 | Score 0.194
------------------------------------------------------------
CONSOLIDATED 
FINANCIAL 
STATEMENTS 
AND INTERNAL 
CONTROL 
REPORTS
56 
IFC 2024 ANNUAL REPORT FINANCIALS
CHUNK 2 | Score 0.194
------------------------------------------------------------
58 
IFC 2024 ANNUAL REPORT FINANCIALS
CHUNK 3 | Score 0.2053
------------------------------------------------------------
MANAGEMENT’S 
DISCUSSION 
AND ANALYSIS
2 
IFC 2024 ANNUAL REPORT FINANCIALS
CHUNK 4 | Score 0.2092
------------------------------------------------------------
This Management’s Discussion and Analysis (MD&A) discusses the financial results of the International Finance 
Corporation (IFC or the Corporation) for the fiscal year ended June 30, 2024 (FY24). The MD&A contains forward- 
looking statements which may be identified by such terms as “anticipates”, “believes”, “expects”, “intends”, “plans”, 
“aims” or words of similar meaning. Such statements involve a number of assumpt

In [25]:
show_chunks("What was IFC's Net Income for the fiscal year ending June 30, 2024?")
# 0.174

08:39:36 | Retrieval | INFO | Starting retrieval process.
08:39:37 | Retrieval | INFO | Found 20 chunks meeting the distance threshold (<= 0.6).
08:39:37 | Retrieval | INFO | Successfully post-processed 20 chunks. Best score: 0.174


Znaleziono chunków: 20

CHUNK 1 | Score 0.174
------------------------------------------------------------
IFC’s net income or loss for the past three fiscal years ended June 30, 2024 are presented below:
Figure 25: IFC's Net Income (Loss) FY22–FY24 
(US$ in millions)
-1,000
0
500
-500
(464)
672
1,000
1,500
2,000
Fiscal year ended June 30,
FY22
FY24
FY23
672
1,485
The following paragraphs detail significant variances between FY24 and FY23 covering the periods included in IFC’s 
FY24 consolidated financial statements. The $813 million increase in net income was principally a result of the following:
Figure 26: Change in Net Income FY24 vs FY23 
(US$ in millions)
298
8
7
48
(97)
(90)
619
(4)
813
24
Equity
Income
URG(L)** 
on Loans 
and Debt
Securities
URG(L)** 
on Equity 
Investments
URG(L)** on 
Borrowings
Loss 
Provision
Net
Treasury
Income**
Administrative
Expenses &
Pensions
Others***
Change in
Net Income
Total Income
from Loans
and Debt
   Securities*
*	
Total income from loans and 

In [26]:
show_chunks("What is the stated mission of the International Finance Corporation?")
# 0.2795

08:39:41 | Retrieval | INFO | Starting retrieval process.
08:39:42 | Retrieval | INFO | Found 20 chunks meeting the distance threshold (<= 0.6).
08:39:42 | Retrieval | INFO | Successfully post-processed 20 chunks. Best score: 0.2272


Znaleziono chunków: 20

CHUNK 1 | Score 0.2272
------------------------------------------------------------
The International Finance Corporation (IFC), an international organization, was established in 1956 to further economic development in 
its member countries by encouraging the growth of private enterprise. IFC is a member of the World Bank Group (WBG), which also 
comprises the International Bank for Reconstruction and Development (IBRD), the International Development Association (IDA), the 
Multilateral Investment Guarantee Agency (MIGA), and the International Centre for Settlement of Investment Disputes (ICSID). Each 
member is legally and financially independent. Transactions with other World Bank Group members are disclosed in the notes that 
follow. IFC’s activities are closely coordinated with and complement the overall development objectives of the other World Bank Group 
institutions. IFC, together with private investors, assists in financing the establishment, improvemen

In [ ]:
show_chunks("What was the revised subscription deadline for the General Capital Increase (GCI) following the April 2024 extension?")
# 0.2913

08:38:05 | Retrieval | INFO | Starting retrieval process.
08:38:06 | Retrieval | INFO | Found 1 chunks meeting the distance threshold (<= 0.3).
08:38:06 | Retrieval | INFO | Successfully post-processed 1 chunks. Best score: 0.2913


Znaleziono chunków: 1

CHUNK 1 | Score 0.2913
------------------------------------------------------------
Following the Spring Meetings in April 2018, a financing package was endorsed by the Board of Governors that included a three-step 
capital raising process: conversion of a portion of retained earnings into paid-in capital, a Selective Capital Increase (SCI) and General 
Capital Increase (GCI) that would provide up to $5.5 billion in additional paid-in capital. The GCI and SCI Resolutions were adopted and 
became effective on April 16, 2020. In April 2023 the deadline for the SCI subscription and payment was extended to April 16, 2025. 
The GCI subscription deadline, initially extended to April 16, 2024 in April 2023, was further extended to April 16, 2025 in April 2024, 
thereby aligning with the original deadline for GCI payment.
The authorized capital stock was increased by the creation of 16,999,998 additional shares each having a par value of $1,000 after


In [ ]:
show_chunks("With which accounting principles do IFC's consolidated financial statements conform?")
# 0.2175

08:40:23 | Retrieval | INFO | Starting retrieval process.
08:40:25 | Retrieval | INFO | Found 20 chunks meeting the distance threshold (<= 0.6).
08:40:25 | Retrieval | INFO | Successfully post-processed 20 chunks. Best score: 0.2175


Znaleziono chunków: 20

CHUNK 1 | Score 0.2175
------------------------------------------------------------
finance and mobilization, IFC offers an array of financial and technical advisory services to private businesses in the developing world 
to increase their chances of success. It also advises governments on how to create an environment hospitable to the growth of private 
enterprise and foreign investment.
NOTE A – SUMMARY OF SIGNIFICANT ACCOUNTING POLICIES
The consolidated financial statements include the financial statements of IFC and its consolidated entities. The accounting and 
reporting policies of IFC conform with accounting principles generally accepted in the United States of America (U.S. GAAP). In the 
opinion of management, the consolidated financial statements reflect all adjustments necessary for the fair presentation of IFC’s 
financial position and results of operations in accordance with U.S. GAAP.
Consolidated Financial Statements presentation – Certain amounts

In [ ]:
show_chunks("What were IFC's Total Long-Term Finance Own Account Commitments in FY24, and what was the corresponding Total disbursed investment portfolio for the Africa region as of June 30, 2024, based on investment portfolio details?")
# 0.2202, 0.2296

08:42:56 | Retrieval | INFO | Starting retrieval process.
08:42:58 | Retrieval | INFO | Found 20 chunks meeting the distance threshold (<= 0.6).
08:42:58 | Retrieval | INFO | Successfully post-processed 20 chunks. Best score: 0.2168


Znaleziono chunków: 20

CHUNK 1 | Score 0.2168
------------------------------------------------------------
STATEMENT OF CUMULATIVE  
GROSS COMMITMENTS  
(at June 30, 2024)
Cumulative Commitments1 (US$ thousands)
Investment Portfolio
REGION
COUNTRY
 NUMBER OF 
ENTERPRISES 
IFC
LOAN & GUARANTEE 
PARTICIPATIONS
TOTAL
Africa
Algeria
15
303,557
5,557
309,114
Angola
12
1,115,087
0
1,115,087
Benin
16
382,514
0
382,514
Botswana
9
102,952
0
102,952
Burkina Faso
33
1,541,083
0
1,541,083
Burundi
11
68,648
7,500
76,148
Cabo Verde
9
65,941
0
65,941
Cameroon
47
1,081,199
471,500
1,552,699
Central African Republic
1
9,880
0
9,880
Chad
13
170,314
13,900
184,214
Comoros
1
14,888
0
14,888
Congo, Democratic Republic of
37
807,086
143,800
950,886
Congo, Republic of
12
164,432
25,000
189,432
Côte d'Ivoire
88
1,623,509
175,275
1,798,783
Djibouti
2
9,290
0
9,290
Egypt, Arab Republic of
152
5,922,853
1,133,389
7,056,242
Eritrea
1
949
0
949
Eswatini
9
47,779
0
47,779
Ethiopia
27
982,657
46,336
1,028,993
Gabon

In [ ]:
show_chunks("What was the total value of IFC's assets as of June 30, 2023?", top=100, threshold=1.0)
# no correct context

08:49:06 | Retrieval | INFO | Starting retrieval process.
08:49:08 | Retrieval | INFO | Found 100 chunks meeting the distance threshold (<= 1.6).
08:49:08 | Retrieval | INFO | Successfully post-processed 100 chunks. Best score: 0.216


Znaleziono chunków: 100

CHUNK 1 | Score 0.216
------------------------------------------------------------
Fair value of assets and liabilities
Estimated fair values of IFC’s financial assets and liabilities and off-balance sheet financial instruments at June 30, 2024 and June 30, 
2023 are summarized below: 
June 30, 2024
June 30, 2023
(US$ in millions)
Carrying 
value
Fair 
value
Carrying 
value
Fair 
value
Financial assets
Cash and due from banks, time deposits, trading securities and 
securities purchased under resale agreements and receivable 
for cash collateral pledged
$ 
40,314 
$ 
40,314 
$ 
47,478 
$ 
47,478 
Investments:
Loans at amortized cost, net of reserve against losses
 
34,648 
 
35,711 
 
29,908 
 
30,369 
Loans accounted for at fair value under the Fair Value Option
 
1,789 
 
1,789 
 
1,506 
 
1,506 
Total loans
 
36,437 
 
37,500 
 
31,414 
 
31,875 
Equity investments accounted for at fair value
 
11,121 a  
11,118 
 
10,778 a  
10,776 
Debt securities accounted

| Question | Chunk | Distatnt score |
| :--- | :--- | :--- |
| What is the official title of the financial report for IFC's 2024 fiscal year? | 58 IFC 2024 ANNUAL REPORT FINANCIALS | 0.194 |
| What was IFC's Net Income for the fiscal year ending June 30, 2024? | IFC’s net income or loss for the past three fiscal years ended June 30, 2024 are presented below:<br>Figure 25: IFC's Net Income (Loss) FY22–FY24 (US$ in millions)-1,0000500-500(464)6721,0001,5002,000Fiscal year ended June 30,FY22FY24FY236721,485<br>The following paragraphs detail significant variances between FY24 and FY23 covering the periods included in IFC’s FY24 consolidated financial statements. The $813 million increase in net income was principally a result of the following:<br>Figure 26: Change in Net Income FY24 vs FY23 <br>...<br>REPORTS56 IFC 2024 ANNUAL REPORT FINANCIALS | 0.174 |
| What is the stated mission of the International Finance Corporation? | open only to member countries of IBRD. IFC is not liable for the obligations of the other institutions. IFC’s mission — as one of the WBG entities — is to end extreme poverty and boost shared prosperity on a livable planet. As the private sector arm of the WBG, IFC provides financing and advisory services to support private sector development in developing economies as a key engine of growth in line with good environmental, social and gov-ernance standards. To further support these efforts, the Board and Management have been working on evolving the WBG to better address the scale of development chal-lenges. As part of this evolution, in October 2023, the Board of Governors endorsed the new vision and mission, as well as initiatives to increase impact, modernize the approach to delivery, and increase financing capacity. Since then, the WBG has started the implementation and operationalization of the evolution process to address | 0.2795 |
| What was the revised subscription deadline for the General Capital Increase (GCI) following the April 2024 extension? | Following the Spring Meetings in April 2018, a financing package was endorsed by the Board of Governors that included a three-step capital raising process: conversion of a portion of retained earnings into paid-in capital, a Selective Capital Increase (SCI) and General Capital Increase (GCI) that would provide up to $5.5 billion in additional paid-in capital. The GCI and SCI Resolutions were adopted and became effective on April 16, 2020. In April 2023 the deadline for the SCI subscription and payment was extended to April 16, 2025. The GCI subscription deadline, initially extended to April 16, 2024 in April 2023, was further extended to April 16, 2025 in April 2024, thereby aligning with the original deadline for GCI payment.The authorized capital stock was increased by the creation of 16,999,998 additional shares each having a par value of $1,000 after | 0.2913 |
| With which accounting principles do IFC's consolidated financial statements conform? | finance and mobilization, IFC offers an array of financial and technical advisory services to private businesses in the developing world to increase their chances of success. It also advises governments on how to create an environment hospitable to the growth of private enterprise and foreign investment.<br>NOTE A – SUMMARY OF SIGNIFICANT ACCOUNTING POLICIES<br>The consolidated financial statements include the financial statements of IFC and its consolidated entities. The accounting and reporting policies of IFC conform with accounting principles generally accepted in the United States of America (U.S. GAAP). In the opinion of management, the consolidated financial statements reflect all adjustments necessary for the fair presentation of IFC’s financial position and results of operations in accordance with U.S. GAAP.<br>Consolidated Financial Statements presentation – Certain amounts in prior years have been changed to conform to the current year’s presentation. | 0.2175 |
| What were IFC's Total Long-Term Finance Own Account Commitments in FY24, and what was the corresponding Total disbursed investment portfolio for the Africa region as of June 30, 2024, based on investment portfolio details? | Table 5: Long-Term Finance and Short-Term Finance Commitments (Own Account and Core Mobilization) FOR THE YEAR ENDED JUNE 30 (US$ in millions)<br>2024 \| 2023 \| VARIANCE<br>Loans: $17,822 \| $14,135 \| $3,687<br>Equity investments: 1,722 \| 1,761 \| (39)<br>Guarantees: 1,857 \| 704 \| 1,153<br>Client risk management: 57 \| 77 \| (20)<br>Total Long-Term Finance Own Account Commitments: $21,458 \| $16,677 \| $4,781<br>Long-Term Finance Core Mobilization<br>Syndication: $8,079 \| $6,165 \| $1,914<br>Advisory and upstream: 4,708 \| 3,687 \| 1,021<br>Anchor investment: 4,202 \| 1,580 \| 2,622<br>Trade finance: 1,824 \| 2,163 \| (339)<br>Third-party-managed funds: 2,720 \| 1,142 \| 1,578<br>IFC-managed funds: 16 \| 151<br>Guarantee: 600 \| — \| 600<br>Other: 355 \| 277 \| 78<br>Total Long-Term Finance Core Mobilizationa: $22,504 \| $15,029 \| $7,475<br>Total Long-Term Finance Commitments: $43,962 \| $31,706 \| $12,256<br><br>\| CHUNK2:   INVESTMENT PORTFOLIO<br>IFC’s total disbursed investment portfolio (a non-U.S. GAAP performance measure) was $59.9 billion as of June 30, 2024 ($52.8 billion — June 30, 2023), as presented in the table below:<br>Figure 10: Disbursed Investment Portfolio Distribution by Region (US$ in billions)<br>Asia and Pacific: 18.9 \| 17.1<br>Latin America and the Caribbean, and Europe: 18.8 \| 16.0<br>Africa: 12.1 \| 10.2<br>Middle East, Central Asia, Türkiye, Pakistan, and Afghanistan: 6.7 \| 6.1<br>Other: 3.4 \| 3.5<br><br>Table 7: Disbursed Investment Portfolio FOR THE YEAR ENDED JUNE 30 (US$ in millions)<br>DISBURSED INVESTMENTS \| AS A % OF TOTAL (2024 \| 2023 \| 2024 \| 2023)<br>Loans: $37,726 \| $32,886 \| 63% \| 62%<br>Equity investments: 10,910 \| 10,371 \| 18% \| 20%<br>Debt securities: 11,221 \| 9,556 \| 19% \| 18%<br>Total Investment Disbursements: $59,857 \| $52,813 \| 100% \| 100%<br>IFC’s disbursed investment portfolio is diversified by region. The distribution of the portfolio by geographical region | 0.2202, 0.2296 |
| What was the total value of IFC's assets as of June 30, 2023? | No proper chunk | - |

In the conducted analysis, there was only a single instance where the retrieval component failed to return the chunk with the correct answer. In this specific case, even after disabling the threshold filtering and increasing the number of retrieved chunks to 100, we still did not obtain the correct answer to the question. In all other cases, the correct answer was consistently contained within a chunk with a distance score between 0.17 and 0.3. This indicates that by setting the threshold to 0.3, we will be able to retrieve the correct answer in the vast majority of cases while avoiding sending redundant context to the model.